partner_address_street3/model/partner.py

```diff
@@ -17,3 +17,12 @@ def _address_fields(self):
         fields = super(ResPartner, self)._address_fields()
         fields.append('street3')
         return fields
+
+    @api.multi
+    def _display_address(self, without_company=False):
+        """Remove empty lines which can happen when street3 field is empty."""
+        res = super(ResPartner, self)._display_address(
+            without_company=without_company)
+        while '\n\n' in res:
+            res = res.replace('\n\n', '\n')
```

Comment:
I think a regexp would do a better job here.

Reply:
@Yajo I changed it to regex